### PR TITLE
Add support for rule categories and ability to exclude categories in config

### PR DESCRIPTION
**Pull request by jeremydelacruz**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our [guidelines](https://github.com/get-woke/woke/blob/main/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)
Link to issue: https://github.com/inclusive-dev-tools/woke/issues/2
Rule definitions currently have no option for assigning and filtering on categories in the config file schema

**What is the new behavior (if this is a feature change)?**
Rule definitions are able to specify optional string categories under the `options` field in the config file. Users are also able to exclude any number of categories with the new `exclude_categories` top-level yaml field.

```yaml
# in this example, both rule1 and rule2 are excluded
exclude_categories:
  - cat1
  - cat2

rules:
  - name: rule1
    terms:
      - rule1
    alternatives:
      - alt-rule1
    severity: warning
    options:
      categories:
        - cat1

  - name: rule2
    terms:
      - rule2
      - rule-2
    alternatives:
      - alt-rule2
      - alt-rule-2
    options:
      categories:
        - cat1
        - cat2

  - name: rule3
    terms:
      - rule3
      - rule-3
    alternatives:
      - alt-rule3
      - alt-rule-3
```
A given rule is able to be assigned any number of categories and a given category is able to be assigned to any number of rules. When a category is "excluded", all rules that contain that category are removed from the enabled ruleset during the rules configuration step.

**Does this PR introduce a breaking change?** (What changes might users need to make due to this PR?)
No breaking changes, categories are totally optional and backwards compatible

**Other information**:

### CODE REVIEW

**Comment by jeremydelacruz** [comment]
Something to note, I see no merge conflicts with the fork's main branch, but saw a potential merge conflict alert when pointed to the original repo -- do we need to fast-forward our main branch again?

**Review by mkcomer — APPROVED** [review]
Looks good to me! Reviewed unit tests & loggers and ran locally. Approved.

File: pkg/config/config.go [hunk 1]
```diff
@@ -67,6 +68,7 @@ func (c *Config) inExistingRules(r *rule.Rule) bool {
 // ConfigureRules adds the config Rules to DefaultRules
 // Configure RegExps for all rules
 // Configure IncludeNote for all rules
+// Filter out any rules that fall under ExcludedCategories
```
Comment (KSLHacks) [inline pkg/config/config.go:71]:
```suggestion
// Filter out any rules that fall under ExcludeCategories
```
nit: property consistency

File: pkg/config/config.go [hunk 2]
```diff
@@ -75,11 +77,40 @@ func (c *Config) ConfigureRules() {
...
 	}
+
+	// Remove excluded rules after done iterating through them
+	if len(c.ExcludeCategories) > 0 {
+		log.Debug().Strs("categories", c.ExcludeCategories).Msg("excluding categories")
+	}
+	for i, exIdx := range excludedIndices {
+		// every time a rule is removed, index of rules that come after it must be reduced by one
+		adjustedIdx := exIdx - i
+		log.Debug().Strs("categories", c.Rules[adjustedIdx].Options.Categories).Msg(fmt.Sprintf("rule \"%s\" excluded with categories", c.Rules[adjustedIdx].Name))
```
Comment (KSLHacks) [inline pkg/config/config.go:103]: nit: format - Is this possible to format? line 103 is very long. Possibly newline indent at `.Msg`?

**Review by KSLHacks — APPROVED** [review]
Looks good! Reviewed in person as well. Few suggested nits to consider.

**Pull request closed without merging**